src/common/charconv.h

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2019 by XGBoost Contributors
+ *
+ * \brief Implement `std::to_chars` and `std::from_chars` for float.  Only base 10 with
+ *        scientific format is supported.  The implementation guarantees roundtrip
+ *        reproducibility.
+ */
+#ifndef XGBOOST_COMMON_CHARCONV_H_
+#define XGBOOST_COMMON_CHARCONV_H_
+
+#include <cstddef>
+#include <system_error>
+#include <iterator>
+#include <limits>
+
+#include "xgboost/logging.h"
+
+namespace xgboost {
+int32_t ToCharsFloatImpl(float f, char * const result);
+
+struct to_chars_result {  // NOLINT
+  char* ptr;
+  std::errc ec;
+};
+
+template <typename T>
+struct NumericLimits;
```

Comment:
Are all of these functions/classes needed in the xgboost namespace? Can you document the functions meant to be used in other files and hide the rest in a detail namespace or some other way?

Reply:
I hid most of the functions into detail namespace.  Sadly the limit has to be available to others as I plan to use `to_chars` and `from_chars` in other places like model dump.  The required buffer size wasn't specified by the C++ ISO, so we have to define it.